department-of-veterans-affairs/abd-vro
# BIP Evidence API fixes

## What was the problem?


BIP Evidence API integration was not complete. There were https issues.

Associated tickets or Slack threads:

- [MCP-000](https://amida.atlassian.net/browse/MCP-000)

## How does this fix it?[^1]


1. This adds a Mock BIP Evidence Service to test https connection + multipart form payload.
2. The application is fixed to test it with the BIP Mock Evidence API (https connection + multiple for payload) 

## How to test this PR
- Build VRO. Make sure it builds and end to end tests are passing
- Build VRO with env=DEV. 
- Set BIP_EVIDENCE_URL to localhost:8094
- Set BIP_KEYSTORE_and BIP_TRUSTSTORE to Base64 values in mock-bip-ce-api client-keystore.yml and truststore.yml.
- In abd-vro ./gradlew :mock-bip-ce-api:bootRun to run the mock server
- Use swagger to send to a file to bip upload end point
-  In mock-bip-ce-api log you should see a location where incoming file is stored
- Make sure the save file is identical to what you sent.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
